python-backend/app/api/documents/variances/resources/variance_document_uploads.py

```diff
@@ -0,0 +1,137 @@
+import base64
+import logging
+import requests
+import json
+from werkzeug.exceptions import BadRequest, NotFound
+
+from flask import request, current_app, Response
+from flask_restplus import Resource, reqparse
+from sqlalchemy.exc import DataError
+
+from ..models.variance import VarianceDocument
+from ...mines.models.mine_document import MineDocument
+
+from app.extensions import api, db
+from ....utils.access_decorators import requires_any_of, MINE_CREATE, MINESPACE_PROPONENT
+from ....utils.resources_mixins import UserMixin, ErrorMixin
+
+
+class VarianceDocumentUploadResource(Resource, UserMixin, ErrorMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument('mine_document_guid', type=str)
+    parser.add_argument('document_manager_guid', type=str)
+    parser.add_argument('filename', type=str)
+    parser.add_argument('mine_guid', type=str)
+    parser.add_argument('mine_no', type=str)
+
+    @api.doc(
+        params={
+            'mine_guid':
+            'Required: The guid of the mine to which the variance will be associated.'
+        })
+    @requires_any_of([MINE_CREATE, MINESPACE_PROPONENT])
+    def post(self):
+        data = self.parser.parse_args()
+        mine_guid = data.get('mine_guid')
+        mine_no = data.get('mine_no')
+
+        if not mine_guid:
+            raise BadRequest('Missing mine_guid')
+
+        if not mine_no:
+            raise BadRequest('Missing mine_no')
```

Comment:
mines live in this microservice (model), as long as we have one of these, it should probably figure out the rest and ensure it exists.